#!/usr/bin/env python
PACKAGE = "image_proc_chain"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("kernel_x",       int_t,    0, "Size x of kernel", 21,  0, 100)
gen.add("kernel_y",       int_t,    0, "Size y of kernel", 21,  0, 100)
gen.add("sigma_x", double_t,    0, "Sigma x",         1.,  0,   10)
gen.add("sigma_y", double_t,    0, "Sigma y",         1.,  0,   10)
gen.add("iteration_count", int_t, 0, "How many process pass through", 1,  1,   100)

border_enum = gen.enum([ gen.const("BorderDefault",      int_t, 0, "A small constant")],
                     "Border processing type")

gen.add("border_type", int_t, 0, "A size parameter which is edited via an enum", 0, 0, 0, edit_method=border_enum)

exit(gen.generate(PACKAGE, "image_proc_chain", "Gaussian"))
